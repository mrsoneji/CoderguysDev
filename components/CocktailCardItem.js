import React from 'react'

// react native-related components
import { View, LogBox, StyleSheet, Text, TouchableHighlight } from 'react-native'
import AsyncImage from '../components/AsyncImage'
import CachedImage from 'react-native-expo-cached-image'

export default class CocktailCardItem extends React.Component {
    constructor(props) {
        super(props);
    
        LogBox.ignoreAllLogs();
    }

    render() {
        const { 
            title = '',
            backgroundColor = 'red',
            imageUri = '',
            id = '',
            onPress = () => {}
        } = this.props

        return (
            <TouchableHighlight onPress={ () => onPress({ id, title }) } style={ [ styles.container, { backgroundColor } ] }>
                <View style={ styles.innerContainer }>
                    <CachedImage source={{ uri: imageUri } } style={[ styles.image ]} />
                    <Text style={ styles.text }>{ title }</Text>
                </View>
            </TouchableHighlight>
        )
    }
}

const styles = StyleSheet.create({
    container: { 
        width: '48%',
        margin: '1%',
        height: 100,
        aspectRatio: 1,
        borderRadius: 8,
        borderColor: '#ffffff',
        borderWidth: 0,
        shadowColor: "#000000",
        shadowOpacity: 0.5,
        shadowRadius: 2,
        shadowOffset: {
            height: 1,
            width: 1
        }
    },
    innerContainer: {
        height: 162,
    },
    text: {
        color: '#000000',
        fontSize: 15,
        fontWeight: 'bold',
        marginTop: 18,
        marginLeft: 8,
        width: 100
    },
    image: {
        height: '100%',
        width: '100%',
        position: 'absolute',
        top: 0,
        left: 0,
        resizeMode: 'cover',
        borderRadius: 8,
        opacity: 0.4,
    }
})
