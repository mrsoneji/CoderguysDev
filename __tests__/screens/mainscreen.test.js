import React from 'react'
import renderer from 'react-test-renderer'
import CocktailCardItem from '../../components/CocktailCardItem'
import CachedImage from 'react-native-expo-cached-image'

describe('CocktailCardItem', () => {
  test('renders correctly', () => {
    jest.mock('react-native-expo-cached-image', () => 'Image')
    const tree = renderer.create(<CocktailCardItem title={ 'Pina Colada' } />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
