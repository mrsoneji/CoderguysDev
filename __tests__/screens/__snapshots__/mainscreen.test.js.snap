// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CocktailCardItem renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "aspectRatio": 1,
        "borderColor": "#ffffff",
        "borderRadius": 8,
        "borderWidth": 0,
        "height": 100,
        "margin": "1%",
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 2,
        "width": "48%",
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <Image
    source={
      Object {
        "cache": "force-cache",
        "uri": "",
      }
    }
    style={
      Array [
        Object {
          "borderRadius": 8,
          "height": "100%",
          "left": 0,
          "opacity": 0.4,
          "position": "absolute",
          "resizeMode": "cover",
          "top": 0,
          "width": "100%",
        },
      ]
    }
  />
  <Text
    style={
      Object {
        "color": "#000000",
        "fontSize": 15,
        "fontWeight": "bold",
        "marginLeft": 8,
        "marginTop": 18,
        "width": 100,
      }
    }
  >
    Pina Colada
  </Text>
</View>
`;
